lib/newrelic_riak/riak_client.rb

```diff
@@ -12,66 +13,84 @@
   end
 
   executes do
-    backend_tracers = proc do
-      add_method_tracer :ping, 'Database/Riak/ping'
 
-      add_method_tracer :list_buckets, 'Database/Riak/list_buckets'
-      add_method_tracer :get_bucket_props, 'Database/Riak/get_bucket_props'
-      add_method_tracer :set_bucket_props, 'Database/Riak/set_bucket_props'
+    # Make this bad boy re-usable
+    backend_tracers = Proc.new do |klass|
+      NewRelic::Agent::Datastores.trace klass, :ping
```

Comment:
Based on the [NR docs](https://docs.newrelic.com/docs/agents/ruby-agent/frameworks/third-party-instrumentation), should these lines all be as follows?

```
NewRelic::Agent::Datastores.trace klass, :ping, 'Riak'
```